basyx.submodelrepository/basyx.submodelrepository-feature-operation-delegation/src/main/java/org/eclipse/digitaltwin/basyx/submodelrepository/feature/operation/delegation/OperationDelegationSubmodelRepository.java

```diff
@@ -0,0 +1,179 @@
+
+	public OperationDelegationSubmodelRepository(SubmodelRepository decorated, OperationDelegation operationDelegation) {
+		this.decorated = decorated;
+		this.operationDelegation = operationDelegation;
+	}
+
+	@Override
+	public CursorResult<List<Submodel>> getAllSubmodels(PaginationInfo pInfo) {
+		return decorated.getAllSubmodels(pInfo);
+	}
+
+	@Override
+	public Submodel getSubmodel(String submodelId) throws ElementDoesNotExistException {
+		return decorated.getSubmodel(submodelId);
+	}
+
+	@Override
+	public void updateSubmodel(String submodelId, Submodel submodel) throws ElementDoesNotExistException {
+		decorated.updateSubmodel(submodelId, submodel);
+	}
+
+	@Override
+	public void createSubmodel(Submodel submodel) throws CollidingIdentifierException, MissingIdentifierException {
+		decorated.createSubmodel(submodel);
+	}
+
+	@Override
+	public void updateSubmodelElement(String submodelIdentifier, String idShortPath, SubmodelElement submodelElement) throws ElementDoesNotExistException {
+		decorated.updateSubmodelElement(submodelIdentifier, idShortPath, submodelElement);
+	}
+
+	@Override
+	public void deleteSubmodel(String submodelId) throws ElementDoesNotExistException {
+		decorated.deleteSubmodel(submodelId);
+	}
+
+	@Override
+	public CursorResult<List<SubmodelElement>> getSubmodelElements(String submodelId, PaginationInfo pInfo) throws ElementDoesNotExistException {
+		return decorated.getSubmodelElements(submodelId, pInfo);
+	}
+
+	@Override
+	public SubmodelElement getSubmodelElement(String submodelId, String smeIdShort) throws ElementDoesNotExistException {
+		return decorated.getSubmodelElement(submodelId, smeIdShort);
+	}
+
+	@Override
+	public SubmodelElementValue getSubmodelElementValue(String submodelId, String smeIdShort) throws ElementDoesNotExistException {
+		return decorated.getSubmodelElementValue(submodelId, smeIdShort);
+	}
+
+	@Override
+	public void setSubmodelElementValue(String submodelId, String smeIdShort, SubmodelElementValue value) throws ElementDoesNotExistException {
+		decorated.setSubmodelElementValue(submodelId, smeIdShort, value);
+	}
+
+	@Override
+	public void createSubmodelElement(String submodelId, SubmodelElement smElement) {
+		decorated.createSubmodelElement(submodelId, smElement);
+	}
+
+	@Override
+	public void createSubmodelElement(String submodelId, String idShortPath, SubmodelElement smElement) throws ElementDoesNotExistException {
+		decorated.createSubmodelElement(submodelId, idShortPath, smElement);
+	}
+
+	@Override
+	public void deleteSubmodelElement(String submodelId, String idShortPath) throws ElementDoesNotExistException {
+		decorated.deleteSubmodelElement(submodelId, idShortPath);
+	}
+
+	@Override
+	public OperationVariable[] invokeOperation(String submodelId, String idShortPath, OperationVariable[] input) throws ElementDoesNotExistException {
+		SubmodelElement sme = getSubmodelElement(submodelId, idShortPath);
+		
+		if (!(sme instanceof InvokableOperation))
```

Comment:
If there is no qualifier, just call decorated.invokeOperation(..)

Reply:
Refactored this as per your suggestion.